.grit/patterns/js/react_legacy_component.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Update React legacy component
+---
+
+detected Legacy component lifecycle
+- `componentWillMount` -> `componentDidMount`  
+- `componentWillReceiveProps` -> `componentDidUpdate` 
+-  `componentWillUpdate` -> `componentDidUpdate`
```

Comment:
These aren't direct replacements, so ideally we have some logic to detect things like when getSnapshotBeforeUpdate should be used. See https://legacy.reactjs.org/docs/react-component.html#unsafe_componentwillupdate.